### PR TITLE
refactor the com.ibm.ws.jaxrs.2.0.web project

### PR DESCRIPTION
**Pull request by WhiteCat22**
to simplify and clean it up so that we can use a single bundle for every version of jaxrs without a hard dependency on jaxrs-2.0

### CODE REVIEW

**Comment by WhiteCat22** [comment]
#build

File: dev/com.ibm.ws.jaxrs.web/src/com/ibm/ws/jaxrs/web/JaxRsInjectionClassListCollaborator.java [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+ * A WebAppInjectionClassListCollaborator is required to tell CDI which application classes require injection.
+ */
+@Component(name = "com.ibm.ws.jaxrs.web.JaxRsInjectionClassListCollaborator",
+           service = WebAppInjectionClassListCollaborator.class,
+           immediate = true,
+           property = { "service.vendor=IBM" })
+public class JaxRsInjectionClassListCollaborator implements WebAppInjectionClassListCollaborator {
+
+    /** {@inheritDoc} */
+    @Override
```
Comment (WhiteCat22) [inline dev/com.ibm.ws.jaxrs.web/src/com/ibm/ws/jaxrs/web/JaxRsInjectionClassListCollaborator.java:46]: This is the only used method in the original class

File: dev/com.ibm.ws.jaxrs.web/src/com/ibm/ws/jaxrs/web/JaxRsInjectionClassListCollaborator.java [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+            if (!isWebModule(moduleContainer)) {
+                return Collections.<String> emptyList();
+            }
+            return getJaxRsInjectionClasses(moduleContainer);
+        } catch (UnableToAdaptException e) {
+            return Collections.<String> emptyList();
+        }
+
+    }
+
```
Comment (WhiteCat22) [inline dev/com.ibm.ws.jaxrs.web/src/com/ibm/ws/jaxrs/web/JaxRsInjectionClassListCollaborator.java:60]: This method is copied from JaxRsUtils

File: dev/com.ibm.ws.jaxrs.web/src/com/ibm/ws/jaxrs/web/JaxRsInjectionClassListCollaborator.java [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+    /**
+     * Answer the JAX-RS injection classes for a web module.
+     *
+     * @param moduleContainer The web module container in which to
+     *                            locate JAX-RS injection classes
+     *
+     * @return The list of JAX-RS injection classes of the
+     *         web module.
+     */
```
Comment (WhiteCat22) [inline dev/com.ibm.ws.jaxrs.web/src/com/ibm/ws/jaxrs/web/JaxRsInjectionClassListCollaborator.java:103]: This method is copied from InjectionUtils

File: dev/com.ibm.ws.jaxrs.web/src/com/ibm/ws/jaxrs/web/JaxRsInjectionClassListCollaborator.java [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+@Component(name = "com.ibm.ws.jaxrs.web.JaxRsInjectionClassListCollaborator",
+           service = WebAppInjectionClassListCollaborator.class,
+           immediate = true,
+           property = { "service.vendor=IBM" })
+public class JaxRsInjectionClassListCollaborator implements WebAppInjectionClassListCollaborator {
+
+    /** {@inheritDoc} */
+    @Override
+    public List<String> getInjectionClasses(Container moduleContainer) {
+        System.out.println("JaxRsInjectionClassListCollaborator.getInjectionClasses()");
```
Comment (andymc12) [inline dev/com.ibm.ws.jaxrs.web/src/com/ibm/ws/jaxrs/web/JaxRsInjectionClassListCollaborator.java:48]: Change to Tr debug - or remove.

File: dev/com.ibm.ws.jaxrs.web/src/com/ibm/ws/jaxrs/web/JaxRsInjectionClassListCollaborator.java [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+     * Annotations are detected on the target class regardless of where the
+     * class is present. Annotations on classes in metadata-complete
+     * regions detected. Annotations on classes which are external to a
+     * target module are detected.
+     *
+     * @param classInfo Information for the class which is to be tested.
+     *
+     * @return True or false telling if the class should be selected for
+     *         injection.
+     */
```
Comment (WhiteCat22) [inline dev/com.ibm.ws.jaxrs.web/src/com/ibm/ws/jaxrs/web/JaxRsInjectionClassListCollaborator.java:224]: This method is copied from InjectionUtils for getJaxRsInjectionClasses

File: dev/com.ibm.ws.jaxrs.web/src/com/ibm/ws/jaxrs/web/JaxRsInjectionClassListCollaborator.java (hunk 2 above)
Comment (andymc12) [reply dev/com.ibm.ws.jaxrs.web/src/com/ibm/ws/jaxrs/web/JaxRsInjectionClassListCollaborator.java:60]: For the methods that you copied from JaxRsUtils and InjectionUtils, can we remove them?  Or are they called from other places in the code?

Comment (WhiteCat22) [reply dev/com.ibm.ws.jaxrs.web/src/com/ibm/ws/jaxrs/web/JaxRsInjectionClassListCollaborator.java:60]: not sure at this time

File: dev/com.ibm.ws.jaxrs.web/src/com/ibm/ws/jaxrs/web/JaxRsInjectionClassListCollaborator.java (hunk 4 above)
Comment (WhiteCat22) [reply dev/com.ibm.ws.jaxrs.web/src/com/ibm/ws/jaxrs/web/JaxRsInjectionClassListCollaborator.java:48]: good catch!

File: dev/com.ibm.ws.jaxrs.2.0.cdi/bnd.bnd [hunk 6]
```diff
@@ -42,7 +42,6 @@ instrument.classesExcludes: com/ibm/ws/jaxrs20/cdi/internal/resources/*.class
 	com.ibm.ws.managedobject;version=latest,\
 	com.ibm.ws.adaptable.module;version=latest,\
 	com.ibm.ws.webcontainer;version=latest,\
-	com.ibm.ws.jaxrs.2.0.web;version=latest,\
```
Comment (WhiteCat22) [inline dev/com.ibm.ws.jaxrs.2.0.cdi/bnd.bnd:45]: not needed at compile time, available at runtime via the feature definition.

File: dev/com.ibm.ws.jaxrs.2.0.ejb/bnd.bnd [hunk 7]
```diff
@@ -33,7 +33,6 @@ instrument.classesExcludes: com/ibm/ws/jaxrs20/ejb/internal/resources/*.class
 	org.apache.cxf.cxf-rt-frontend-jaxrs;strategy=exact;version=3.1.18,\
 	com.ibm.ws.jaxrs.2.0.common;version=latest,\
 	com.ibm.ws.jaxrs.2.0.server;version=latest,\
-	com.ibm.ws.jaxrs.2.0.web;version=latest,\
```
Comment (WhiteCat22) [inline dev/com.ibm.ws.jaxrs.2.0.ejb/bnd.bnd:36]: not needed at compile time, available at runtime via the feature definition.

**Comment by WhiteCat22** [comment]
#build

**Comment by WhiteCat22** [comment]
looks like jaxrs compiled fine but Artifactory had a network issue. I removed the moved methods that are no longer being using in InjectionUtils and need another #build anyway

**Comment by andrea-pichler** [comment]
No message review requested from L2

**Comment by WhiteCat22** [comment]
cancelling this PR

**Pull request closed without merging**